api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java

```diff
@@ -535,6 +521,31 @@ private Workspace updateWorkspaceImpl(
     return ResponseEntity.ok(new CloneWorkspaceResponse().workspace(savedWorkspace));
   }
 
+  /**
+   * Gets a FireCloud Billing project.
+   *
+   * <p>If {@code enableFireCloudV2Billing} is enabled, create one directly using FireCloud
+   * v2Billing endpoints. Otherwise, claim one from billing buffer.
+   */
+  private FirecloudWorkspaceId getFcBillingProject(DbAccessTier accessTier, Workspace workspace) {
+    DbUser user = userProvider.get();
+    String billingProject;
+    if (workbenchConfigProvider.get().featureFlags.enableFireCloudV2Billing) {
+      // If v2 Billing is enabled, we will call FireCloud directly to create one.
+      billingProject = billingProjectBufferService.createBillingProjectName();
+      fireCloudService.createAllOfUsBillingProject(
+          billingProject, accessTier.getServicePerimeter());
+
+      // We use AoU Service Account to create the billing account then assign owner role to user.
+      // In this way, we can make sure AoU Service Account is still the owner of this billing
+      // account.
+      fireCloudService.addOwnerToBillingProject(user.getUsername(), billingProject);
```

Comment:
This is correct for parity with the current code, but as a follow-up after PPW - we should definitely consider whether we still need to give them owner. A lower role may suffice now that we're planning to do billing configuration via Terra.